rust/hyperlane-core/src/chain.rs

```diff
@@ -289,30 +242,42 @@ impl Debug for HyperlaneDomain {
 }
 
 impl HyperlaneDomain {
-    pub fn from_config(domain_id: u32, name: &str) -> Result<Self, &'static str> {
+    pub fn from_config(
+        domain_id: u32,
+        name: &str,
+        implementation: HyperlaneDomainImpl,
+    ) -> Result<Self, &'static str> {
+        let name = name.to_ascii_lowercase();
         if let Ok(domain) = KnownHyperlaneDomain::try_from(domain_id) {
-            if name.to_ascii_lowercase().as_str() == domain.as_str() {
+            if name == domain.as_str() {
                 Ok(HyperlaneDomain::Known(domain))
             } else {
                 Err("Chain name does not match the name of a known domain id; the config is probably wrong.")
             }
+        } else if name.as_str().parse::<KnownHyperlaneDomain>().is_ok() {
+            Err("Chain name is known the domain is incorrect; the config is probably wrong.")
         } else {
             Ok(HyperlaneDomain::Unknown {
                 domain_id,
-                chain_name: name.to_owned(),
+                chain_name: name,
                 // we might want to support accepting these from the config later
                 domain_type: HyperlaneDomainType::Unknown,
-                domain_impl: HyperlaneDomainImpl::Unknown,
```

Comment:
Hmm, I'm surprised it worked for me to add an unknown domain via config, given that we set the domain_impl here to Unknown.

Why did this work for me?

Reply:
It works for you because in the current `main` branch version we just assume it is ethereum where it actually matters; this is getting us ready for fuel support.